src/Aspire.Dashboard/ResourceService/Partials.cs

```diff
@@ -48,19 +48,31 @@ public ResourceViewModel ToViewModel(BrowserTimeProvider timeProvider, IKnownPro
                 State = HasState ? State : null,
                 KnownState = HasState ? Enum.TryParse(State, out KnownResourceState knownState) ? knownState : null : null,
                 StateStyle = HasStateStyle ? StateStyle : null,
-                ReadinessState = HasHealthState ? HealthState switch
-                {
-                    HealthStateKind.Healthy => ReadinessState.Ready,
-                    _ => ReadinessState.NotReady,
-                } : ReadinessState.Unknown,
-                Commands = GetCommands()
+                Commands = GetCommands(),
+                HealthReports = HealthReports.Select(ToHealthReportViewModel).ToImmutableArray(),
             };
         }
         catch (Exception ex)
         {
             throw new InvalidOperationException($@"Error converting resource ""{Name}"" to {nameof(ResourceViewModel)}.", ex);
         }
 
+        HealthReportViewModel ToHealthReportViewModel(HealthReport healthReport)
+        {
+            return new HealthReportViewModel(healthReport.Key, Convert(healthReport.Status), healthReport.Description, healthReport.Exception);
+
+            static Microsoft.Extensions.Diagnostics.HealthChecks.HealthStatus Convert(HealthStatus healthStatus)
+            {
+                return healthStatus switch
+                {
+                    HealthStatus.Unknown or HealthStatus.Healthy => Microsoft.Extensions.Diagnostics.HealthChecks.HealthStatus.Healthy,
```

Comment:
Why is unknown state considered healthy? Intentional or mistake?

If it is intentional, could you add a comment saying why.

Reply:
Innocent until proven guilty. Will add a comment.

Reply:
What does wait for do when a dependency has a health check that returns unknown? This should be consistent with it @mitchdenny 

It would be odd to have a green tick with an unknown status in the UI, but wait for is still waiting for it to enter a health state.

Reply:
I looked into this more, and we're using the extensions health check enum on the server and the client (dashboard) with an unknown status inbetween on the proto layer.

It's impossible for the app host to return unknown, but a custom resource service implementation could return it.

I think the dashboard should stop using `Microsoft.Extensions.Diagnostics.HealthChecks.HealthStatus` and use the enum that reflects all the values that can be returned from the server. That way we don't have to guess what is meant by unknown.

Reply:
Alternatively, maybe there shouldn’t be an unknown option in the service layer?

Reply:
I believe the most recent push addresses the concerns here.

Reply:
See 68578f85006decbd721df7d2017a18cf9489ff99 and 14d55aca335107abbe937a52ec2f14921084ec68.